notebooks/MAST/JWST/JWST_EngineeringDatabase_Retreival/JWST_EDB_Retrieval.ipynb

```diff
@@ -0,0 +1,560 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "times = { \n",
+    "         't_start': '20220601T000000',\n",
+    "         't_end':   '20220601T060000'\n",
+    "        }\n",
+    "mnemonics = {\n",
+    "            'SA_ZADUCMDX': times,\n",
+    "            'SA_ZADUCMDY': times\n",
+    "           }\n",
+    "for m,v in mnemonics.items():\n",
+    "    print(m, v)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Construct File Names\n",
+    "\n",
+    "The key to fetching data from the web service is to construct file names to contain the data for each mnemonic. The web service will parse the file names to determine how to query the engineering database and retrieve the timeseries of interest.\n",
+    "\n",
+    "The file names have the form: \n",
+    "\n",
+    "    `<mnemonic_name>-<t_start>-<t_end>.csv`\n",
+    "    \n",
+    "Use a dictionary comprehension to construct a list of file names; these will be passed to the webservice calling function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fnames = ['-'.join([m, v['t_start'], v['t_end']]) + '.csv' for m,v in mnemonics.items()]\n",
+    "print(fnames)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Call the Webservice"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the (optional) output folder name prior to the webservice call. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sub-directory where the data files will be written:\n",
+    "subdir = 'edb_test'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now call the EDB web service. The files containing data will be written to your local storage, in the specified subdirectory. \n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "\n",
+    "<span style=\"color:black\">\n",
+    "    The webservice may take a long time (or timeout), depending upon the quantity of data in the timeseries within the chosen date range. The six-hour time interval in this example returns over 80,000 rows.  \n",
+    "    \n",
+    "</span>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "status = download_edb_datafiles(token, fnames, folder=subdir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare the Data for Analysis\n",
+    "\n",
+    "Create a list of pandas dataframes from the mnemonics data that were just written to disk. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = [pd.read_csv(subdir+'/'+f) for f in fnames]"
```

Comment:
You should use `os.path.sep` instead of assuming '/' for directories, since we want this to be runnable on Windows.